crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -152,7 +152,18 @@ where
     }
 
     fn header_td_by_number(&self, number: BlockNumber) -> ProviderResult<Option<U256>> {
-        self.database.header_td_by_number(number)
+        match self.database.header_td_by_number(number)? {
+            Some(td) => Ok(Some(td)),
+            None => {
+                let block_hash = self.canonical_in_memory_state.state_by_number(number)
+                    .map(|state| state.block().block().hash());
+                if let Some(hash) = block_hash {
+                    self.database.header_td(&hash)
```

Comment:
this is also slightly more nuanced here because these blocks are supposed to be only blocks after the merge block, so in this, we actually need to find the last td entry on disk and use that

Reply:
Hi @mattsse,

I've implemented the changes based on the feedback. Could you please review the latest commit and let me know if there are any further adjustments needed?